docs/sphinx_doc/en/source/tutorial/206-prompt.md

```diff
@@ -1,16 +1,279 @@
+)
+
+prompt = model.format(
+   Msg("system", "You're a helpful assistant", role="system"),   # Msg object
+   [                                                             # a list of Msg objects
+      Msg(name="Bob", content="Hi.", role="assistant"),
+      Msg(name="Alice", content="Nice to meet you!", role="assistant"),
+   ],
+)
+print(prompt)
+```
+
+```bash
+[
+  {"role": "user", "content": "system: You are a helpful assistant\nBob: Hi.\nAlice: Nice to meet you!"},
+]
+```
+
+### `OllamaChatWrapper`
+
+`OllamaChatWrapper` encapsulates the Ollama chat API, which takes a list of
+messages as input. The message must obey the following rules (updated in
+2024/03/22):
+
+- Require `role` and `content` fields, and `role` must be either `"user"`,
+  `"system"`, or `"assistant"`.
+- An optional `images` field can be added to the message
+
+#### Prompt Strategy
+
+Given a list of messages, we will parse each message as follows:
+
+- `Msg`:  Fill the `role` and `content` fields directly. If it has an `url`
+  field, which refers to an image, we will add it to the message.
+- `List`: Parse each element in the list according to the above rules.
+
+```python
+from agentscope.models import OllamaChatWrapper
+
+model = OllamaChatWrapper(
+    config_name="", # empty since we directly initialize the model wrapper
+    model_name="llama2",
+)
+
+prompt = model.format(
+   Msg("system", "You're a helpful assistant", role="system"),   # Msg object
+   [                                                             # a list of Msg objects
+      Msg(name="Bob", content="Hi.", role="assistant"),
+      Msg(name="Alice", content="Nice to meet you!", role="assistant", url="https://example.com/image.jpg"),
+   ],
+)
+
+print(prompt)
+```
+
+```bash
+[
+  {"role": "system", "content": "You are a helpful assistant"},
+  {"role": "assistant", "content": "Hi."},
+  {"role": "assistant", "content": "Nice to meet you!", "images": ["https://example.com/image.jpg"]},
+]
+```
+
+### `OllamaGenerationWrapper`
+
+`OllamaGenerationWrapper` encapsulates the Ollama generation API, which
+takes a string prompt as input without any constraints (updated to 2024/03/22).
+
+#### Prompt Strategy
+
+We will ignore the `role` field and combine the prompt into a single string
+of conversation.
+
+```python
+from agentscope.models import OllamaGenerationWrapper
+from agentscope.message import Msg
+
+model = OllamaGenerationWrapper(
+    config_name="", # empty since we directly initialize the model wrapper
+    model_name="llama2",
+)
+
+prompt = model.format(
+   Msg("system", "You're a helpful assistant", role="system"),   # Msg object
+   [                                                             # a list of Msg objects
+      Msg(name="Bob", content="Hi.", role="assistant"),
+      Msg(name="Alice", content="Nice to meet you!", role="assistant"),
+   ],
+)
+
+print(prompt)
+```
+
+```bash
+system: You are a helpful assistant
+Bob: Hi.
+Alice: Nice to meet you!
+```
```

Comment:
seems missing Ollama embedding model (which is already implemented but not mentioned here)

Reply:
The prompt strategies are only for chat and generation API. For embedding and multi-modal API, the developers can use them directly.